app/src/main/java/com/android/feedme/model/viewmodel/HomeViewModel.kt

```diff
@@ -0,0 +1,160 @@
+package com.android.feedme.model.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.android.feedme.model.data.Profile
+import com.android.feedme.model.data.ProfileRepository
+import com.android.feedme.model.data.Recipe
+import com.android.feedme.model.data.RecipeRepository
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class HomeViewModel : ViewModel() {
+
+  private val recipeRepository = RecipeRepository.instance
+  private val profileRepository = ProfileRepository.instance
+
+  private val _recommendedRecipes = MutableStateFlow<List<Recipe>>(emptyList())
+  private val _filteredRecipes = MutableStateFlow<List<Recipe>>(emptyList())
+
+  private val _recipes = MutableStateFlow<List<Recipe>>(emptyList())
+  val recipes = _recipes.asStateFlow()
+
+  private val _filteredProfiles = MutableStateFlow<List<Profile>>(emptyList())
+  val filteredProfiles = _filteredProfiles.asStateFlow()
+
+  var initialSearchQuery = ""
+
+  init {
+    FirebaseAuth.getInstance().uid?.let {
+      fetchRecipe("lasagna1")
+      fetchRecipe("lasagna1")
+      fetchRecipe("lasagna1")
+    }
+  }
+
+  /**
+   * A function that fetches the recipe during Login
+   *
+   * @param id: the unique ID of the recipe we want to fetch
+   */
+  fun fetchRecipe(id: String) {
+    viewModelScope.launch {
+      recipeRepository.getRecipe(
+          id,
+          onSuccess = { recipe ->
+            if (recipe != null) {
+              _recommendedRecipes.value += recipe
+              _recipes.value = _recommendedRecipes.value
+            }
+          },
+          onFailure = {
+            // Handle failure
+            throw error("Recipe was not fetched during Login")
+          })
+    }
+  }
+
+  /**
+   * A function that fetches the recipes given their Ids
+   *
+   * @param ids: the unique IDs of the profiles we want to fetch
+   */
+  /*fun fetchRecipes(ids: List<String>) {
```

Comment:
What is exactly this commented code ?

Reply:
This method was present before I started working on the view model.
I believe it's a method that will be modified in a later issue in order to fetch multiple recipes at the same time (in 1 query). It will be used when we will fetch the base recipes in Home page (it's written below for the recommendation algorithm).

Reply:
ok thanks for the clarification ! Otherwise really good job !! Thank youuu :D